### PR TITLE
Add augment_plus and augment_minus to PrivateKey

### PR DESCRIPTION
**Pull request by coltfred**
See #94 

TODO 

- [x] tests showing proper Fr augmentation.

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -1061,6 +1061,30 @@ impl PrivateKey {
...
+    ///just using `+` if you need to rely on the order of the curve and the mod
+    ///on the private keys lining up.
+    pub fn augment_plus(&self, other: &PrivateKey) -> PrivateKey {
+        PrivateKey::augment(self, other, true)
+    }
+    ///Augment the private key with another. This function should be preferred over
+    ///just using `-` if you need to rely on the order of the curve and the mod
+    ///on the private keys lining up.
+    pub fn augment_minus(&self, other: &PrivateKey) -> PrivateKey {
+        PrivateKey::augment(self, other, true)
```
Comment (BobWall23) [inline src/api.rs]: This looks curiously like `augment_plus`

File: src/api_480.rs [hunk 2]
```diff
@@ -1095,6 +1095,30 @@ impl PrivateKey {
...
+    ///just using `+` if you need to rely on the order of the curve and the mod
+    ///on the private keys lining up.
+    pub fn augment_plus(&self, other: &PrivateKey) -> PrivateKey {
+        PrivateKey::augment(self, other, true)
+    }
+    ///Augment the private key with another. This function should be preferred over
+    ///just using `-` if you need to rely on the order of the curve and the mod
+    ///on the private keys lining up.
+    pub fn augment_minus(&self, other: &PrivateKey) -> PrivateKey {
+        PrivateKey::augment(self, other, true)
```
Comment (BobWall23) [inline src/api_480.rs]: Ditto

File: src/api.rs (hunk 1 above)
Comment (coltfred) [reply src/api.rs]: Not anymore! Tests!

File: src/api_480.rs (hunk 2 above)
Comment (coltfred) [reply src/api_480.rs]: Tests!

**Comment by clintfred** [comment]
Did you consider introducing a generic parameter onto `PrivateKey` to handle the `Fp256` vs `Fr256`. An advantage would be that `+` and `-` could be separately defined for `PrivateKey<Fp256>` vs `PrivateKey<Fr256>`.
Another potential solution would be to make two separate types.

**Comment by coltfred** [comment]
@clintfred Previously to hide the underlying `FpXXX` types from the public api. If we didn't want to do that still we could do what you're suggesting. I would argue that there is limited use of doing that and it just pushes more information onto the caller than they care about.

One could argue that we should remove `-` and `+` from the public apis as they have limited value to a caller.

**Comment by clintfred** [comment]
I agree. I would suggest we remove `+` and `-` from PrivateKey until we have a use for it. It seems like it will serve to confuse users, and if we ever needed it again it's simple to add back in.

Unless someone else has a usecase for it that I'm not thinking of...

**Pull request merged**